pkg/client/interface.go

```diff
@@ -313,3 +314,20 @@ type SharedQueryClient interface {
 type BlockQueryClient interface {
 	Block(ctx context.Context, height *int64) (*coretypes.ResultBlock, error)
 }
+
+// ProofParams is a go interface type which corresponds to the poktroll.proof.Params
+// protobuf message. Since the generated go types don't include interface types, this
+// is necessary to prevent dependency cycles.
```

Comment:
#PUC what the dependency cycle is.

Reply:
I think this will need to become its own issue where we evaluate potential solutions. There are potential dependency cycles looming in `pkg/client` due to the importing of concrete module types. In this case, we were trying to import `x/proof/types` for the `Params` type but that package imports `pkg/client` for some querier interface types.

Reply:
https://github.com/pokt-network/poktroll/issues/664